### PR TITLE
feat: add tip feature

### PR DESCRIPTION
**Pull request by justinenerio**
## Changes

- Add tipping USDC via QR code

## Videos
<details><summary>Create Tip </summary>

https://user-images.githubusercontent.com/28533130/206650089-91fcc52d-5614-4f87-8c5e-b4297c2a9dcc.mp4

</details>

<details><summary>Recieve Tip (Reciever)</summary>

https://user-images.githubusercontent.com/28533130/206650442-fef2bbe9-fa56-4b94-86a4-490f27387e81.mp4

</details>

<details><summary>Recieve Tip (Sender)</summary>
(can skip to end to see update)

https://user-images.githubusercontent.com/28533130/206650270-b1859df1-3e97-42fa-b531-1a47410316e7.mp4

</details>

## Checklist

- [x] PR is ready for review (if not, it should be a draft).
- [x] PR title follows [Conventional Commits][1] guidelines.
- [x] Screenshots/video added.
- [x] Tests added.
- [x] Self-review done.

[1]: https://www.conventionalcommits.org/en/v1.0.0/

### CODE REVIEW

**Review by ookami-kb — CHANGES_REQUESTED** [review]
You have dependency cycles, please remove them:

```
/app/screens/authenticated/wallet_flow/wallet_flow_screen.dart -> /app/screens/authenticated/wallet_flow/wallet_main_screen.dart -> /app/screens/authenticated/wallet_flow/wallet_flow_screen.dart
/features/incoming_tip_payments/src/presentation/link_listener.dart -> /features/incoming_tip_payments/module.dart -> /features/incoming_tip_payments/src/presentation/link_listener.dart
```

You can check them by running `make deps_graph_all deps_cycles`.

**Comment by justinenerio** [comment]
@ookami-kb have removed the dependency cycles

File: packages/cryptoplease/lib/features/outgoing_tip_payments/src/bl/bloc.dart [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    required Ed25519HDKeyPair escrow,
+    required Token token,
+  }) async {
+    final privateKey = await escrow.extract().then((value) => value.bytes.lock);
+    final key = base58encode(privateKey.toList());
+
+    final link = TipPaymentData(
+      key: key,
+      token: token.publicKey,
+    ).toUri();
```
Comment (ookami-kb) [inline packages/cryptoplease/lib/features/outgoing_tip_payments/src/bl/bloc.dart:193]: The link should be wrapped into Firebase deep link (and internal QR code scanner should be able to process it as well).

**Review by ookami-kb — CHANGES_REQUESTED** [review]

**Comment by justinenerio** [comment]
@ookami-kb updated to deeplink and tests

**Review by ookami-kb — APPROVED** [review]

**Pull request merged**